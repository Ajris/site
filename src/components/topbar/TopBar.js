import React from 'react'
import {Image, Nav, Navbar} from "react-bootstrap";
import logo from "./logo.PNG";
import TopBarElement from "./TopBarElement";
import TopBarLoginElement from "./TopBarLoginElement";
import * as Constants from "../../constants/Constants";

export default class TopBar extends React.Component {
    render() {
        return (
            <Navbar collapseOnSelect expand="lg" bg="dark" variant="dark" sticky='top'>
                <Navbar.Brand href={Constants.HOME_URL}>
                    <div>
                        <Image roundedCircle fluid src={logo} alt={"logo"} style={{width: '40px', height: '40px'}}
                               className={"mx-auto"}/>
                    </div>
                </Navbar.Brand>
                <Navbar.Toggle aria-controls="responsive-navbar-nav"/>
                <Navbar.Collapse id="responsive-navbar-nav">
                    <Nav className="mr-auto" variant="tabs" fill justify>
                        <TopBarElement name={Constants.HOME_NAME_COMPONENT}/>
                        <TopBarElement name={Constants.EXPERIENCE_NAME_COMPONENT}/>
                        <TopBarElement name={Constants.PROJECTS_NAME_COMPONENT}/>
                        <TopBarElement name={Constants.TECHNOLOGIES_NAME_COMPONENT}/>
                    </Nav>
                    <Nav variant="tabs" fill justify>
                        <TopBarLoginElement name={Constants.LOGIN_NAME_COMPONENT}/>
                    </Nav>
                </Navbar.Collapse>
            </Navbar>)
    }
}