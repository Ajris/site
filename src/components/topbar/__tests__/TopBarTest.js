import React from "react";
import { create } from "react-test-renderer";
import TopBar from "../TopBar";
import {BrowserRouter} from "react-router-dom";

describe("TopBar component", () => {
    test("it matches the snapshot", () => {
        const component = create(
            <BrowserRouter>
                <TopBar/>
            </BrowserRouter>);
        expect(component.toJSON()).toMatchSnapshot();
    });
});