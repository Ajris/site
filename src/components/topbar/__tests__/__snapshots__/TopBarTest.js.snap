// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBar component it matches the snapshot 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-dark bg-dark sticky-top"
>
  <a
    className="navbar-brand"
    href="/"
  >
    <div>
      <img
        alt="logo"
        className="mx-auto img-fluid rounded-circle"
        src="logo.PNG"
        style={
          Object {
            "height": "50px",
            "width": "50px",
          }
        }
      />
    </div>
  </a>
  <button
    aria-controls="responsive-navbar-nav"
    aria-label="Toggle navigation"
    className="navbar-toggler"
    onClick={[Function]}
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    aria-expanded={null}
    className="navbar-collapse collapse"
    id="responsive-navbar-nav"
  >
    <div
      className="mr-auto container bordersOut navbar-nav"
      onKeyDown={[Function]}
    >
      <a
        className="nav-link"
        onClick={[Function]}
      >
        <div
          className="nav-item"
        >
          About
        </div>
      </a>
      <a
        className="nav-link"
        onClick={[Function]}
      >
        <div
          className="nav-item"
        >
          Blog
        </div>
      </a>
      <a
        className="nav-link"
        onClick={[Function]}
      >
        <div
          className="nav-item"
        >
          Projects
        </div>
      </a>
      <a
        className="nav-link"
        onClick={[Function]}
      >
        <div
          className="nav-item"
        >
          Technologies
        </div>
      </a>
    </div>
  </div>
</nav>
`;
